azkaban-common/src/main/java/azkaban/imagemgmt/rampup/ImageRampupManagerImpl.java

```diff
@@ -134,6 +135,29 @@ public ImageRampupManagerImpl(final ImageRampupDao imageRampupDao,
     return imageTypeVersionMap;
   }
 
+  @Override
+  public Map<String, VersionInfo> validateAndGetUpdatedVersionMap(final VersionSet versionSet)
+      throws ImageMgmtException {
+    // Find the image types for which version is either invalid or not exists
+    final Set<String> imageTypesWithInvalidVersion = versionSet.getImageToVersionMap().entrySet()
```

Comment:
How expensive this call is? We are calling invalid version check for each image type.

Reply:
Yes for each image type invalid version check query is called. The image version check query is boolean query and hence not expensive. However, we need to add index on the appropriate column.

Reply:
Sounds good. I have added ticket for the same.